cmd/docker/docker.go

```diff
@@ -44,7 +51,23 @@ func newDockerCommand(dockerCli *command.DockerCli) *cli.TopLevelCommand {
 			return fmt.Errorf("docker: '%s' is not a docker command.\nSee 'docker --help'", args[0])
 		},
 		PersistentPreRunE: func(cmd *cobra.Command, args []string) error {
-			return isSupported(cmd, dockerCli)
+			if err := isSupported(cmd, dockerCli); err != nil {
+				return err
+			}
+
+			name := cmd.Name()
+			for p := cmd.Parent(); p != nil && p != cmd.Root(); p = p.Parent() {
+				name = p.Name() + " " + name
+			}
+
+			ctx, _ := otel.Tracer("").Start(cmd.Context(), name)
+			cmd.SetContext(ctx)
+			_ = dockerCli.Apply(command.WithContext(ctx))
+			return nil
+		},
+		PersistentPostRun: func(cmd *cobra.Command, args []string) {
+			// TODO: There doesn't seem to be a way to determine if the command returned an error, so we can set the span status here.
+			trace.SpanFromContext(cmd.Context()).End()
```

Comment:
You're fighting an uphill trying to use cobra lifecycle hooks for this. See below.